# **(ADD THE NOTEBOOK NAME HERE)**

In [6]:
import os

# Go up one folder from the notebook's location
os.chdir('..')

# Show the current working directory
print("Current working directory:", os.getcwd())

Current working directory: c:\Users\ramoz\OneDrive\Desktop\my-carprices-project\The-car-prices


In [2]:
import os 
print(os.listdir())

['.git', '.gitignore', '.python-version', '.slugignore', '.venv', 'Dataset', 'jupyter_notebooks', 'Procfile', 'README.md', 'requirements.txt', 'setup.sh']


In [7]:
from pathlib import Path
import pandas as pd

# Path to your CSV file
RAW_FILE = Path("Dataset/Raw/CarPrice_Assignment (2).csv")

# Load the CSV
print("📄 Using raw file:", RAW_FILE)
df_raw = pd.read_csv(RAW_FILE)

# Show shape and first few rows
print("✅ Loaded shape:", df_raw.shape)
df_raw.head()

📄 Using raw file: Dataset\Raw\CarPrice_Assignment (2).csv
✅ Loaded shape: (205, 26)


car_ID  symboling                   CarName fueltype aspiration doornumber  \
0       1          3        alfa-romero giulia      gas        std        two   
1       2          3       alfa-romero stelvio      gas        std        two   
2       3          1  alfa-romero Quadrifoglio      gas        std        two   
3       4          2               audi 100 ls      gas        std       four   
4       5          2                audi 100ls      gas        std       four   

       carbody drivewheel enginelocation  wheelbase  ...  enginesize  \
0  convertible        rwd          front       88.6  ...         130   
1  convertible        rwd          front       88.6  ...         130   
2    hatchback        rwd          front       94.5  ...         152   
3        sedan        fwd          front       99.8  ...         109   
4        sedan        4wd          front       99.4  ...         136   

   fuelsystem  boreratio  stroke compressionratio horsepower  peakrpm citympg  \
0        mpfi       3.47    2.68              9.0        111     5000      21   
1        mpfi       3.47    2.68              9.0        111     5000      21   
2        mpfi       2.68    3.47              9.0        154     5000      19   
3        mpfi       3.19    3.40             10.0        102     5500      24   
4        mpfi       3.19    3.40              8.0        115     5500      18   

   highwaympg    price  
0          27  13495.0  
1          27  16500.0  
2          26  16500.0  
3          30  13950.0  
4          22  17450.0  

[5 rows x 26 columns]

## Objectives

* Write your notebook objective here, for example, "Fetch data from Kaggle and save as raw data", or "engineer features for modelling"

## Inputs

* Write down which data or information you need to run the notebook 

## Outputs

* Write here which files, code or artefacts you generate by the end of the notebook 

## Additional Comments

* If you have any additional comments that don't fit in the previous bullets, please state them here. 



In [8]:
df_raw.shape
df_raw.head()
df_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 205 entries, 0 to 204
Data columns (total 26 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   car_ID            205 non-null    int64  
 1   symboling         205 non-null    int64  
 2   CarName           205 non-null    object 
 3   fueltype          205 non-null    object 
 4   aspiration        205 non-null    object 
 5   doornumber        205 non-null    object 
 6   carbody           205 non-null    object 
 7   drivewheel        205 non-null    object 
 8   enginelocation    205 non-null    object 
 9   wheelbase         205 non-null    float64
 10  carlength         205 non-null    float64
 11  carwidth          205 non-null    float64
 12  carheight         205 non-null    float64
 13  curbweight        205 non-null    int64  
 14  enginetype        205 non-null    object 
 15  cylindernumber    205 non-null    object 
 16  enginesize        205 non-null    int64  
 1

---

In [10]:
# Remove duplicates
df_raw = df_raw.drop_duplicates()

# Handle missing values
df_raw = df_raw.fillna(0)  # or df_raw.dropna()

# Rename columns (optional)
df_raw.columns = df_raw.columns.str.strip().str.lower().str.replace(' ', '_')

# Example: Create a new feature
df_raw['price_per_engine_size'] = df_raw['price'] / df_raw['enginesize']

# Change working directory

In [11]:
clean_path = Path("Dataset/clean/car_prices_clean.csv")
df_raw.to_csv(clean_path, index=False)
print("✅ Cleaned file saved to:", clean_path)


✅ Cleaned file saved to: Dataset\clean\car_prices_clean.csv


* We are assuming you will store the notebooks in a subfolder, therefore when running the notebook in the editor, you will need to change the working directory

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [12]:
import os
current_dir = os.getcwd()
current_dir

'c:\\Users\\ramoz\\OneDrive\\Desktop\\my-carprices-project\\The-car-prices'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [13]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [14]:
current_dir = os.getcwd()
current_dir

'c:\\Users\\ramoz\\OneDrive\\Desktop\\my-carprices-project'

# Section 1

This code is to use the raw dataset and shape the data (205 rows and 26 columns)
from pathlib import Path
import pandas as pd

# Path to your CSV file
RAW_FILE = Path("Dataset/Raw/CarPrice_Assignment (2).csv")

# Load the CSV
print("📄 Using raw file:", RAW_FILE)
df_raw = pd.read_csv(RAW_FILE)

# Show shape and first few rows
print("✅ Loaded shape:", df_raw.shape)
df_raw.head()


Section 1 content

from pathlib import Path
import pandas as pd

# Path to your CSV file
RAW_FILE = Path("Dataset/Raw/CarPrice_Assignment (2).csv")

# Load the CSV
print("📄 Using raw file:", RAW_FILE)
df_raw = pd.read_csv(RAW_FILE)

# Show shape and first few rows
print("✅ Loaded shape:", df_raw.shape)
df_raw.head()

---

# Section 2

Section 2 content

the following code, shows the data shape, display the head (first 5 rows) and it shows the summary of the dataset columns names, data type... etc  
df_raw.shape
df_raw.head()
df_raw.info()

This codes gives the following output:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 205 entries, 0 to 204
Data columns (total 26 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   car_ID            205 non-null    int64  
 1   symboling         205 non-null    int64  
 2   CarName           205 non-null    object 
 3   fueltype          205 non-null    object 
 4   aspiration        205 non-null    object 
 5   doornumber        205 non-null    object 
 6   carbody           205 non-null    object 
 7   drivewheel        205 non-null    object 
 8   enginelocation    205 non-null    object 
 9   wheelbase         205 non-null    float64
 10  carlength         205 non-null    float64
 11  carwidth          205 non-null    float64
 12  carheight         205 non-null    float64
 13  curbweight        205 non-null    int64  
 14  enginetype        205 non-null    object 
 15  cylindernumber    205 non-null    object 
 16  enginesize        205 non-null    int64  
 17  fuelsystem        205 non-null    object 
 18  boreratio         205 non-null    float64
 19  stroke            205 non-null    float64
...
 24  highwaympg        205 non-null    int64  
 25  price             205 non-null    float64
dtypes: float64(8), int64(8), object(10)
memory usage: 41.8+ KB

#Section 3


NOTE

* You may add as many sections as you want, as long as it supports your project workflow.
* All notebook's cells should be run top-down (you can't create a dynamic wherein a given point you need to go back to a previous cell to execute some task, like go back to a previous cell and refresh a variable content)

---

# Push files to Repo

* In cases where you don't need to push files to Repo, you may replace this section with "Conclusions and Next Steps" and state your conclusions and next steps.

In [ ]:
import os
try:
  # create your folder here
   os.makedirs(name='')
except Exception as e:
  print(e)


IndentationError: expected an indented block after 'try' statement on line 2 (553063055.py, line 5)